src/snowflake/cli/api/exceptions.py

```diff
@@ -16,6 +16,14 @@ class MissingConfiguration(ClickException):
     pass
 
 
+class CycleDetectedError(ClickException):
```

Comment:
nit: feels a little out of place here, in the sense that this exception is rather specific to one part of the logic, but I defer to the CLI team ultimately for this one.

Reply:
If usage of the exception is self-contained to a module / package I would move the exception there.